airflow/providers/amazon/aws/utils/eks_kube_config.py

```diff
@@ -0,0 +1,133 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+from shutil import which
+from typing import Optional
+
+import boto3
+import yaml
+
+HOME = os.environ.get('HOME', '/tmp')
+DEFAULT_KUBE_CONFIG_FILENAME = 'config'
+DEFAULT_KUBE_CONFIG_PATH = str(os.path.join(HOME, '/.kube/', DEFAULT_KUBE_CONFIG_FILENAME))
+DEFAULT_CONTEXT_NAME = 'aws'
+DEFAULT_NAMESPACE_NAME = 'default'
+DEFAULT_POD_USERNAME = 'aws'
+
+
+def generate_config_file(
```

Comment:
This appears to be used only in one place, EKSPodOperator, so this possibly doesn't need to be in a separate file

Perhaps this could be a method on the EksHook? (Given it connects to EKS and does describe cluster, I think that is the right place for it)

Reply:
Yes, it is only used in the EKSPodOperator.  I had kept it to a separate utils file because it felt cleaner to me and the Hooks file was/is strictly one API endpoint, one method.  I'm happy to adjust if you feel the EKSHook class is a better home.

Reply:
Moved into the operator class in a coming revision.

Reply:
Addressed in https://github.com/apache/airflow/pull/16571/commits/2606556296bfeaccabb513397f05a3885ac44431